### PR TITLE
fix(docs): broken internal links in the query docs

### PR DESCRIPTION
**Pull request by fulopkovacs**
I found 100+ broken links starting with `/query/v5/docs` with [broken-link-checker](https://www.npmjs.com/package/broken-link-checker). This PR fixes them.

Most of these come from moving the `core` pages away from the `framework`-related pages.

# TODO

Fix the links for:
- [x] React
- [x] Angular
- [x] Vue
- [x] Solid
- [x] Svelte 

# How I tested the links

1. Started the docs site locally on `http://localhost:3000` (https://github.com/TanStack/tanstack.com)
2. Ran the following commands for each framework in the local clone of this repo (this example is for angular):
```bash
# Check the site for broken links: print the results in the terminal and save them to this file: `broken-links-angular.txt`
broken-link-checker http://localhost:3000/query/v5/docs/framework/angular/overview  -roe --filter-level 0 --requests 20 | tee broken-links-angular.txt
# Find all the relevant broken links in the `broken-links-angular.txt` file
grep -E '.*BROKEN.*http://localhost:3000/query/v5' broken-links-angular.txt
```
These commands can be chained if you don't need the output in the terminal. For Angular, this is how the output should look like when all the links are fixed (the `injectQuery.md` page is supposed to be missing, since it'll be added in the future):

image

<details>
  <summary>Testing the broken links with a script</summary>
   
```bash
#!/usr/bin/env bash
# Check all the broken links for the React Query docs (v5)
# NOTE: You must install the `broken-link-checker` package from npm
# NOTE: You must run the `tanstack/tanstack.com` site locally on port 3000
# NOTE: This script will run for a loooong time :'(, but in the end it should
# print all the links that are broken in the `v5` docs

# 1. React
broken-link-checker http://localhost:3000/query/v5/docs/framework/react/overview  -roe --filter-level 0 --requests 20 | grep -E '.*BROKEN.*http://localhost:3000/query/v5'

# 2. Vue
broken-link-checker http://localhost:3000/query/v5/docs/framework/vue/overview  -roe --filter-level 0 --requests 20 | grep -E '.*BROKEN.*http://localhost:3000/query/v5'

# 3. Angular
broken-link-checker http://localhost:3000/query/v5/docs/framework/angular/overview  -roe --filter-level 0 --requests 20 | grep -E '.*BROKEN.*http://localhost:3000/query/v5'

# 4. Solid
broken-link-checker http://localhost:3000/query/v5/docs/framework/solid/overview  -roe --filter-level 0 --requests 20 | grep -E '.*BROKEN.*http://localhost:3000/query/v5'

# 5. Svelte
broken-link-checker http://localhost:3000/query/v5/docs/framework/svelte/overview  -roe --filter-level 0 --requests 20 | grep -E '.*BROKEN.*http://localhost:3000/query/v5'

```
</details>

### CODE REVIEW

**Comment by SeanCassiere** [comment]
@fulopkovacs Found some other ones in #7248, do you mind including them in yours?

**Comment by fulopkovacs** [comment]
> @fulopkovacs Found some other ones in #7248, do you mind including them in yours?

Oh yeah, sure, thanks for your work! This PR contains like <10% of all the broken links, so I still have a lot more links to fix! 😢

**Comment by fulopkovacs** [comment]
@TkDodo I couldn't figure out where should the `../injectQuery` link go to here: https://github.com/TanStack/query/blob/b11c876bbdb9cf344daa74c1bb5ca2b15c0bcd2e/docs/framework/angular/guides/caching.md?plain=1#L26

So that one is still broken.

**Comment by TkDodo** [comment]
@fulopkovacs in the react docs, this links to the `useQuery` reference:

https://tanstack.com/query/v5/docs/framework/react/reference/useQuery

in angular, `useQuery` is `injectQuery`. However, there don't seem to be dedicated API docs for angular (yet?)

I think we can either remove the link or point it towards the same location as `useQuery`, but named `injectQuery` so that it will work once the site is created.

@arnoud-dv FYI

**Comment by arnoud-dv** [comment]
> or point it towards the same location as useQuery, but named injectQuery so that it will work once the site is created.

That would be my preference, I certainly plan to add the API reference.

**Comment by fulopkovacs** [comment]
> > or point it towards the same location as useQuery, but named injectQuery so that it will work once the site is created.
> 
> That would be my preference, I certainly plan to add the API reference.

Awesome, thanks for the responses. I updated the link to point at `https://tanstack.com/query/v5/docs/framework/angular/reference/injectQuery`.

https://github.com/fulopkovacs/query/blob/b8547375b8844c6b607386f7723a4db536f915ef/docs/framework/angular/guides/caching.md?plain=1#L26

**Comment by fulopkovacs** [comment]
😅 Ugh, I checked back on this PR, and it looks like I only checked the links for `react`, so I'll fix the links for the rest of the frameworks too.

After that, I'll update the description of this PR with the terminal commands I used (spoiler: [`broken-link-checker`](https://www.npmjs.com/package/broken-link-checker) and `grep`), and that'll enable others to review this work properly.

**Comment by fulopkovacs** [comment]
@TkDodo I think I'm ready with this PR, could you check it please? (The script I used to check the broken links is in the description, under the accordion.)

Feel free to tell me if something is missing.

**Comment by fulopkovacs** [comment]
Sad status update: I ran into some issues as I started adding more complex tests. This is a bummer because I wanted to get this one ready during the weekend. 😢

File: docs/framework/react/guides/important-defaults.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Out of the box, TanStack Query is configured with **aggressive but sane** defaul
 
 Have a look at the following articles from our Community Resources for further explanations of the defaults:
 
-- [Practical React Query](../tkdodos-blog#1-practical-react-query)
-- [React Query as a State Manager](../tkdodos-blog#10-react-query-as-a-state-manager)
+- [Practical React Query](../../community/tkdodos-blog#1-practical-react-query)
+- [React Query as a State Manager](../../community/tkdodos-blog#10-react-query-as-a-state-manager)
```
Comment (TkDodo) [inline docs/framework/react/guides/important-defaults.md:40]: looking at the file structure, we are in:

`docs/framework/react/guides/important-defaults.md`

and we want to link towards

`docs/framework/react/community/tkdodos-blog.md`

so imo the correct link should be one level up:

`../community/tkdodos-blog`

this is also what my editor would auto-complete. So why are two levels up (`../../` ) needed ?

Comment (fulopkovacs) [reply docs/framework/react/guides/important-defaults.md:40]: I'm not sure why the two `..`-s are needed. I just assumed that it's related to the conversation in the [in the `#router-maintainers` channel in Discord](https://discord.com/channels/719702312431386674/1015406599738761307/1227323140925816986) about trailing slashes, and [VSCode improperly flagging links with errors](https://discord.com/channels/719702312431386674/1015406599738761307/1227343543748530217).  Also, it did not work the VSCode-way. 🤣 (Still doesn't, I checked.)

I'll ask around in Discord, and come back with the answer.

Comment (fulopkovacs) [reply docs/framework/react/guides/important-defaults.md:40]: Okay, I presented this theory to Tanner, and he said [it's correct](https://discord.com/channels/719702312431386674/1015406599738761307/1229555107473920131): 

1. the file is `docs/framework/react/guides/important-defaults.md`
2. we want to link to `docs/framework/react/community/tkdodos-blog.md`
3. the relative link that works is `../../community/tkdods.md`

My theory on why we need two `..`-s here:
- somewhere along the way the original markdown gets interpreted as the `docs/framework/react/guides/important-defaults/` route
- `.` parent is `important-defaults`
- `..` grandparent is `guides`
- `../..` great-grandparent is `react`

Hence `./../../community` takes us to the `community` route in the current path's great-grandparent route.

**Review by TkDodo — APPROVED** [review]

**Pull request merged**